datest/run.go

```diff
@@ -0,0 +1,167 @@
+package datest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+	"encoding/json"
+
+	"github.com/commonjava/indy-tests/common"
+)
+
+type Report struct {
+	ExecutionRoot struct {
+		GroupID     string `json:"groupId"`
+		ArtifactID  string `json:"artifactId"`
+		Version     string `json:"version"`
+		OriginalGAV string `json:"originalGAV"`
+	} `json:"executionRoot"`
+
+	Modules []struct {
+		Gav struct {
+			GroupID     string `json:"groupId"`
+			ArtifactID  string `json:"artifactId"`
+			Version     string `json:"version"`
+			OriginalGAV string `json:"originalGAV"`
+		} `json:"gav"`
+		ManagedPlugins struct {
+		} `json:"managedPlugins"`
+		ManagedDependencies struct {
+			Dependencies map[string]struct{
+				GroupID     string `json:"groupId"`
+				ArtifactID  string `json:"artifactId"`
+				Version     string `json:"version"`
+			} `json:"dependencies"`
+		} `json:"managedDependencies,omitempty"`
+	} `json:"modules"`
+}
+
+func getAlignLog(url string) string {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		panic(err)
+	}
+	req.Header.Set("Accept", "text/plain")
+
+	var c http.Client
+	resp, err := c.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	responseData, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(responseData)
+}
+
+func lookupMetadata(url string) string {
+	fmt.Println(url)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		panic(err)
+	}
+	req.Header.Set("Accept", "application/xml")
+
+	var c http.Client
+	resp, err := c.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	bodyString := string(bodyBytes)
+
+	if strings.Contains(bodyString, "Message:") {
+		fmt.Printf(bodyString)
+	}
+
+	return "Done"
+}
+
+func Run(targetIndy, daGroup string, processNum int) {
+
+	indyHost, validated := common.ValidateTargetIndy(targetIndy)
+	if !validated {
+		os.Exit(1)
+	}
+
+	indyURL := "http://" + indyHost
+
+	routines := processNum
+
+	var urls []string
+
+	files, err := ioutil.ReadDir("alignment_report/")
+    if err != nil {
+        log.Fatal(err)
+    }
+ 
```

Comment:
Recommend to use "go fmt" command to format all code before commit, or use vscode+go plugin to format them automatically.